src/xinterpreter.cpp

```diff
@@ -240,15 +240,60 @@ namespace xpyt
     void interpreter::redirect_display()
     {
         py::module sys = py::module::import("sys");
+        py::module py_json = py::module::import("json");
 
         py::module xeus_python_display = py::module::import("xeus_python_display");
         m_displayhook = xeus_python_display.attr("XPythonDisplay")();
 
-        py::cpp_function publish_display = [this](int execution_counter, py::object obj){
+        py::cpp_function publish_display = [this, py_json](int execution_counter, py::object obj){
             if (!obj.is_none())
             {
                 xeus::xjson pub_data;
-                pub_data["text/plain"] = py::str(obj);
+
+                if (hasattr(obj, "_repr_mimebundle_"))
```

Comment:
TODO: separate in get_display_pub_data (or other name) function, to be called both from the display hook and the explicit `display` and `display_id` functions.